// SPDX-FileCopyrightText: 2021 Mario Franke <research@m-franke.net>
//
// SPDX-License-Identifier: GPL-2.0-or-later

syntax = "proto2";

import "skyfield_protobuf/simulation.proto";
import "skyfield_protobuf/satellite.proto";

package skyfield_protobuf;

message Message {
    optional uint32 id = 1;

    oneof message_type {
        satellite.Message satellite = 100;
        simulation.Message simulation = 200;
        Acknowledge ack = 300;
    }
}

message Acknowledge {
    optional uint32 id = 1; // used to acknowledge a message with identical id
}
